plugin/pkg/auth/authenticator/token/bootstrap/bootstrap.go

```diff
@@ -184,3 +192,28 @@ func parseToken(s string) (string, string, error) {
 	}
 	return split[1], split[2], nil
 }
+
+// getGroups loads and validates the bootstrapapi.BootstrapTokenGroupsKey key
+// from the bootstrap token secret, returning a list of group names or an error
+// if any of the group names are invalid.
+func getGroups(secret *api.Secret) ([]string, error) {
+	groupsString := getSecretString(secret, bootstrapapi.BootstrapTokenGroupsKey)
+	if groupsString == "" {
+		return []string{bootstrapapi.BootstrapDefaultGroup}, nil
+	}
+
+	// split on commas amd validate the prefix on each group name
+	groups := strings.Split(groupsString, ",")
+	for _, group := range groups {
+		if group == bootstrapapi.BootstrapDefaultGroup {
+			// allow the default group even though it doesn't strictly match the prefix
+			continue
+		}
+
+		if !strings.HasPrefix(group, bootstrapapi.BootstrapGroupPrefix) {
```

Comment:
We don't do any kind of validation like this with other authenticators. It seems more sane to either keep these arbitrary strings or define some some subset of characters the group can use than defend against some possible injections. If you don't trust the value you should sanitize it anyway and shouldn't trust a regexp.